src/core/ddsi/src/q_entity.c

```diff
@@ -4583,12 +4681,24 @@ static void create_proxy_builtin_endpoints(
       assert (is_builtin_entityid (guid1.entityid, proxypp->vendor));
       if (is_writer_entityid (guid1.entityid))
       {
+        // Replace topic name for each pwr
```

Comment:
It seems to me that it would make more sense to refactor it a bit and pass topic & type name in as arguments.